relay-crons/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+//! Crons protocol and processing for Sentry.
+//!
+//! [Cron Monitors] allow you to monitor the uptime and performance of any scheduled, recurring job
+//! in Sentry. Once implemented, it'll allow you to get alerts and metrics to help you solve errors,
+//! detect timeouts, and prevent disruptions to your service.
+//!
+//! # API
+//!
+//! The public API documentation is available on [Sentry Docs](https://docs.sentry.io/api/crons/).
+//!
+//! [cron monitors]: https://docs.sentry.io/product/crons/
+
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png",
+    html_favicon_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png"
+)]
+#![warn(missing_docs)]
+
+use relay_common::Uuid;
+use serde::{Deserialize, Serialize};
+
+/// Error returned from [`process_checkin`].
+#[derive(Debug, thiserror::Error)]
+pub enum ProcessCheckinError {
+    /// Failed to deserialize the payload.
+    #[error("failed to deserialize checkin")]
+    Json(#[from] serde_json::Error),
+}
+
+///
+#[derive(Clone, Copy, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+enum CheckinStatus {
```

Comment:
Could we name it 
```suggestion
enum CheckInStatus {
```

in basically all the instances of it, in docs as well? 

Or rename it to just `Check`

Reply:
`CheckIn` is what we have been using the api

Reply:
In the protocol it's `checkin_id` and also the URL is `checkin`, so I think either we go with the clumsier `check_in_id` for these cases or we embrace `Checkin`.